src/t_set.c

```diff
@@ -804,16 +804,17 @@ void spopWithCountCommand(client *c) {
         return;
     }
 
-    /* Case 2 and 3 require to replicate SPOP as a set of SREM commands.
+    /* Case 2 and 3 require to replicate SPOP as a SREM command.
      * Prepare our replication argument vector. Also send the array length
      * which is common to both the code paths. */
-    robj *propargv[3];
-    propargv[0] = shared.srem;
-    propargv[1] = c->argv[1];
+    unsigned long batchsize = 1024;
+    unsigned long propremian = count % batchsize == 0 ? 0 : 1;
+    unsigned long batch = count / batchsize;
+    robj **propargv = zmalloc(sizeof(robj *) * (2 * (batch + propremian) + count));
```

Comment:
I prefer to alloc an argv* array which has the length of `2 + (count > batchsize ? count : batchsize)`, and propagate everytime the array is filled, so that we can avoid creating and holding too many objects at the same time.

Reply:
I also prefer just having one buffer we fill up and propagate periodically. I also think we should stack allocate the batch buffer. I also think we should add a small helper that adds the item to the buffer, propagates, and frees as needed.